Rework Quick & Helm Installation Guide

tgraf opened the pull request:
Major rework of quick & helm install guides. Most guides are consolidated into two main ones.

Rendered preview:
https://deploy-preview-15695--cilium-doc.netlify.app/

errordeveloper commented on Documentation/gettingstarted/cni-chaining-azure-cni.rst:
```diff
@@ -39,7 +41,7 @@ example below, the Azure CNI, portmap, and Cilium are chained together.
     metadata:
       name: cni-configuration
-      namespace: cilium
+      namespace: kube-system
```
This looks a little incidental, why was `cilium` namespace used previously? If this has to be changed in this PR, a separate commit would be nice.

ti-mo replied:
I proposed this change earlier, but someone (@aanm?) mentioned there used to be a service (on GKE?) that periodically removed certain resources from `kube-system`. Not sure if this is still the case.

aanm replied:
yes, that was the case on GKE, I'm not sure about AKS.

ti-mo replied:
Alright, should be fine then :+1:

errordeveloper commented on Documentation/gettingstarted/requirements-rke.rst:
```diff
@@ -0,0 +1,28 @@
+
+**Requirements:**
+
+* Follow the `RKE Installation Guide <https://rancher.com/docs/rke/latest/en/installation/>`_
```
I think this needs updating to https://docs.rke2.io/install/network_options/#using-cilium-instead-of-canal (cc @seanmwinn)

errordeveloper commented on Makefile:
```diff
@@ -577,9 +577,6 @@ install-manpages: ## Install manpages the Cilium CLI.
 postcheck: build ## Run Cilium build postcheck (update-cmdref, build documentation etc.).
 	$(QUIET)$(MAKE) $(SUBMAKEOPTS) -C Documentation update-cmdref check
 
-minikube: ## Setup a minikube with Cilium installed.
```
One less make target that has nothing to do with builds, yet does too much - yey! 🎉 

(This probably needs a separate commit, not sure how it relates to docs changes).

errordeveloper replied:
I guess it happen to be an incidental dependency of quick install generation logic?

vadorovsky replied:
It wasn't a dependency of quick install generation logic, but rather a target similar to `microk8s` - it builds cilium container image and then deploys it on minikube. My goal was to have some kind of lightweight alternative to vagrant.

But since it didn't end up being used, I'm fine with removing it. There is still microk8s for similar workflow. And running locally built images on kind is also an option.

errordeveloper commented on Documentation/gettingstarted/k8s-install-sandbox.rst:
```diff
@@ -1,17 +0,0 @@
-    Please use the official rendered version released here:
-    https://docs.cilium.io
-
-Creating a Sandbox environment
```
I can see how this and other sections made sense in theory, but resulted in users having to do quite a few extra clicks.

errordeveloper reviewed (commented):
New guide looks good to me overall from the diff, will need to have a good look at what it looks like in the browser and will probably post some more comment :)

tgraf committed:
k8s: Remove quick-install.yaml generation logic

As we no longer depend on the various quick-install YAML files, we can
remove all the logic to generate them.

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
doc: New quick install guide

This commit reworks the quick installation guides to use the Cilium CLI.
This allows to completely remove the quick-install.yaml.

The Helm installation guides are kept and moved into a new "Advanced
Installation" section.

This also allows to standardize the Observability instructions and move
it into a single guide split into Setup, CLI, and UI.

In order to really simplify, the only remaining guide that was using the
cilium namespace for installation is also changed to use kube-system so
all follow-up guides can be kept the same.

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
Add new unified Helm guide

Consolidate most separate installation guides covering Helm-based
installation into a single guide with tabs for the different platforms.

Some guides are left intact which are either quite special or relatively
uncommon.

Signed-off-by: Thomas Graf <thomas@cilium.io>

jibi commented on Documentation/gettingstarted/k8s-install-helm.rst:
```diff
@@ -0,0 +1,242 @@
+          connectivity to resources outside the cluster (e.g., VMs in the VPC
+          or AWS managed services) is masqueraded (i.e., SNAT) by Cilium to use
+          the VPC IP address of the Kubernetes worker node.  Excluding the
+          lines for ``eni.enabled=true`` and ``tunnel=disabled`` from the helm
```
Based on the tests in https://github.com/cilium/cilium/issues/15190, in order to enable tunneling mode on EKS I think we should exclude also `ipam.mode=eni` from the list of helm flags.

I don't have enough context on `nodeinit.enabled=true` but it appears it's not required as well for tunneling mode

tgraf replied:
``nodeinit.enabled=true`` is needed for now until we port the new init containers model from the CLI over to Helm.

nbusseneau commented on Documentation/gettingstarted/k8s-install-helm.rst:
```diff
@@ -0,0 +1,242 @@
+       In order to allow cilium-operator to interact with the Azure API, a
+       service principal is required. You can reuse an existing service
+       principal if you want but it is recommended to create a dedicated
+       service principal for cilium-operator:
```
I would suggest mentioning this should be done per installation, otherwise one might re-use the same SP for several clusters:
```suggestion
       In order to allow cilium-operator to interact with the Azure API, a
       service principal is required. You can reuse an existing service
       principal if you want but it is recommended to create a dedicated
       service principal for each Cilium installation:
```
But then, we should also warn that if one incidentally uses the same `--name` as an existing SP, the previous SP will be overwritten. Note the `Found an existing application instance of "6316910e-005f-41c6-a432-bfcc7e459548". We will patch it` in this example:
```console
➜ az ad sp create-for-rbac --name nbusseneau-test-overwrite
Changing "nbusseneau-test-overwrite" to a valid URI of "http://nbusseneau-test-overwrite", which is the required format used for service principal names
In a future release, this command will NOT create a 'Contributor' role assignment by default. If needed, use the --role argument to explicitly create a role assignment.
Creating 'Contributor' role assignment under scope '/subscriptions/22716d91-fb67-4a07-ac5f-d36ea49d6167'
The output includes credentials that you must protect. Be sure that you do not include these credentials in your code or check the credentials into your source control. For more information, see https://aka.ms/azadsp-cli
{
  "appId": "6316910e-005f-41c6-a432-bfcc7e459548",
  "displayName": "nbusseneau-test-overwrite",
  "name": "http://nbusseneau-test-overwrite",
  "password": "<a first password>",
  "tenant": "***"
}
➜ az ad sp create-for-rbac --name nbusseneau-test-overwrite
Changing "nbusseneau-test-overwrite" to a valid URI of "http://nbusseneau-test-overwrite", which is the required format used for service principal names
Found an existing application instance of "6316910e-005f-41c6-a432-bfcc7e459548". We will patch it
In a future release, this command will NOT create a 'Contributor' role assignment by default. If needed, use the --role argument to explicitly create a role assignment.
Creating 'Contributor' role assignment under scope '/subscriptions/22716d91-fb67-4a07-ac5f-d36ea49d6167'
  Role assignment already exists.

The output includes credentials that you must protect. Be sure that you do not include these credentials in your code or check the credentials into your source control. For more information, see https://aka.ms/azadsp-cli
{
  "appId": "6316910e-005f-41c6-a432-bfcc7e459548",
  "displayName": "nbusseneau-test-overwrite",
  "name": "http://nbusseneau-test-overwrite",
  "password": "<a new password, replacing the previous one>",
  "tenant": "***"
}
```

To avoid this happening, I would suggest we switch to either:

- Using `--name cilium-operator-$RANDOM`
- Not providing `--name` ([CLI will generate one if not provided](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest#az_ad_sp_create_for_rbac-optional-parameters))

nbusseneau commented on Documentation/gettingstarted/k8s-install-gke.rst:
```diff
@@ -1,104 +0,0 @@
-.. only:: not (epub or latex or html)
```
I think we have some leftovers to `k8s-install-gke` and friends, e.g. in `install/kubernetes/cilium/README.*` and `.github/ISSUE_TEMPLATE/release_template.md`. I suppose we should update them to the new guides as well.

nbusseneau commented on Documentation/gettingstarted/k8s-install-default.rst:
```diff
@@ -4,34 +4,281 @@
+
+.. tabs::
+
+    .. group-tab:: GKE
```
[Nit] For consistency, we should rename tabs for GKE/AKS/EKS to include the platform names, as in the next set of tabs.

tgraf replied:
Unlike the tabs below, these are actually specific to GKE, EKS, AKS, and so on.

nbusseneau replied:
Sorry, I'm not sure I follow 🤔
Does that mean there is a difference between the GKE tab highlighted above and the GCP/GKE tab present in the next set of tabs (line 134 in same file)?
```rst
.. group-tab:: GCP/GKE
```

tgraf replied:
Yes, the "GKE" tab in create cluster is creating a GKE cluster. The "GCP/GKE" tab in "Install Cilium" will function for both GKE and self-managed clusters in GCP. I'm OK to also label GCP above if you think that's less confusing overall.

nbusseneau replied:
Ah I see! I think I got mostly confused by the fact the `GCP/GKE`-like tabs explicitly only mention provider-managed clusters (e.g. `Install Cilium into the GKE cluster:`), so I did not think the docs were targeting self-managed GCP clusters.

Perhaps we ought to rephrase some parts a bit differently for self-managed clusters.

christarazi commented on contrib/scripts/minikube.sh:
```diff
@@ -1,22 +0,0 @@
-
-make docker-images-all DOCKER_IMAGE_TAG=dev
-
-cp "install/kubernetes/quick-install.yaml" /tmp/cilium-minikube.yaml
-
-sed -i 's|latest|dev|g' /tmp/cilium-minikube.yaml
-sed -i 's|docker.io/||g' /tmp/cilium-minikube.yaml
-sed -i 's|imagePullPolicy: Always|imagePullPolicy: Never|g' /tmp/cilium-minikube.yaml
-
-kubectl create -f /tmp/cilium-minikube.yaml
```
Instead of removing this script and the minikube target as there might be developers who depend on it (I'm not sure who though), maybe we can replace the highlighted lines with the cilium-cli?

tgraf replied:
I've asked @mrostecki whether the script is still needed.

vadorovsky replied:
I didn't use the script lately.

The question is if anyone finds it useful. If not, I'm fine with removing it. Otherwise can follow up with switching to cilium-cli in it.

christarazi reviewed (approved):
LGTM, one optional minor comment on minikube target

nbusseneau commented on Documentation/gettingstarted/k8s-install-default.rst:
```diff
@@ -4,34 +4,287 @@
 ==============
 
-.. parsed-literal::
+You can install Cilium on any Kubernetes cluster. If you already have a
+Kubernetes cluster, skip this step. If you want to create a cluster, pick one
+of the options below:
```
Just noticed the bit about skipping the step if a cluster is already available should probably be moved to the previous section.
